plans/bitswap-tuning/test/transfer.go

```diff
@@ -93,37 +93,38 @@ func Transfer(runenv *runtime.RunEnv) error {
 		}
 
 		// Inform other nodes of the root CID
-		if _, err = writer.Write(RootCidSubtree, &rootCid); err != nil {
+		if _, err = writer.Write(ctx, RootCidSubtree, &rootCid); err != nil {
 			return fmt.Errorf("Failed to get Redis Sync RootCidSubtree %w", err)
 		}
 	} else if isLeech {
 		// Get the root CID from a seed
 		rootCidCh := make(chan *cid.Cid, 1)
-		cancelRootCidSub, err := watcher.Subscribe(RootCidSubtree, rootCidCh)
-		if err != nil {
+		sctx, cancelRootCidSub := context.WithCancel(ctx)
+		if err := watcher.Subscribe(sctx, RootCidSubtree, rootCidCh); err != nil {
```

Comment:
We are not calling `cancelRootCidSub` on `error`?  I think it is good to always call cancel, and if we add it as a defer at one place it would be easier to reason about it.